lib/ansible/module_utils/basic.py

```diff
@@ -845,6 +847,7 @@ def __init__(self, argument_spec, bypass_checks=False, no_log=False,
         self.aliases = {}
         self._legal_inputs = ['_ansible_%s' % k for k in PASS_VARS]
         self._options_context = list()
+        self._tmpdir_created = False
```

Comment:
Rather than keeping a boolean here, I'd use a self._remote_tmpdir (which holds the value of tmpdir passed from the controller) and self._tmpdir which holds the full path to the actual tmpdir.).